base/dict.jl

```diff
@@ -344,6 +344,7 @@ ht_keyindex2!(h::Dict, key) = ht_keyindex2_shorthash!(h, key)[1]
     h.vals[index] = v
     h.count += 1
     h.age += 1
+    h.ndel = ifelse(@inbounds(isslotmissing(h, index)), h.ndel - 1, h.ndel)
```

Comment:
Looks to me like the `@inbounds` should be used by the caller of this function (ref how the index access a few lines above does not have `@inbounds` on them.

Reply:
My question was just if the inbounds propagation carries to the next level in the call chain, and without firm understanding of that it felt safer and harmless to add the inbounds here as any bounds violation would have thrown already at this point